### PR TITLE
Introduce ability to freeze weights

### PR DESCRIPTION
**Pull request by WardLT**
Following https://arxiv.org/html/2502.15582v1

Includes code from https://github.com/ACEsuit/mace/pull/830, which I'll eliminate once it's merged.

### CODE REVIEW

**Comment by 7radians** [comment]
Hey, thank you for your interest :) In case this is useful, I've also just pushed the learning rate scaling feature, which acts as a "soft freeze" mode. We assign a low learning rate to some parameters instead of freezing them completely. I will add instructions to the docs on how to use it soon.

**Comment by WardLT** [comment]
That does sound awesome. Thanks, @7radians ! I'll be happy to experiment with it and tell you what I learn. 😁

Short version: I'm fine-tuning MACE-MP for MD solutions of metals dissolved in liquids
